checks/__init__.py

```diff
@@ -962,7 +962,7 @@ def agent_formatter(metric, value, timestamp, tags, hostname, device_name=None,
     """
     attributes = {}
     if tags:
-        attributes['tags'] = list(set(tags))
+        attributes['tags'] = tags
```

Comment:
@truthbk Have you test `event` and `service_check` ? I may be wrong but that may impact them.

Reply:
I checked, it still works !